ProxyAgent/HttpClient/HttpRequest.cs

```diff
@@ -153,8 +155,21 @@ public IHttpRequest SetContent<T>(T sourceObject, Encoding encoding, MediaTypeHe
         public IHttpRequest SetContent(byte[] content, string mediaType)
         {
             this.requestContent.Content = new ByteArrayContent(content);
-            this.ContentType = mediaType == null ? this.defaultMediaType : new MediaTypeHeaderValue(mediaType);
+
+            // When content type is not defined, then assume JSON because that's the typical use for this application.
+            // If the content type is defined, and it's not a file transfer, then set the content type.
+            // For file transfers (via multipart), the content type is set in HttpClient.SetHeaders
+            // using TryAddWithoutValidation. Setting multipart content type here would cause an exception.
+            if (mediaType == null)
+            {
+                this.ContentType = this.defaultMediaType;
+            }
+            else if (!mediaType.StartsWith(MULTIPART_REQUEST_CONTENT_TYPE_PREFIX, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Where are you setting the content type when it is a multipart request?

Reply:
in `HttpClient.cs` (see above comment on line 161)

Reply:
I saw the comment but as you said in the comment it says if the content type is defined and not a file transfer. So when it is a file transfer (multipart/form-data) you don't set content type to anything in this logic from what I can tell. Is that expected and works okay?

Reply:
yes it works, we ran multiple tests with different http request types